rules/no-useless-promise-resolve-reject.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+const {methodCallSelector} = require('./selectors/index.js');
+
+const RETURN_RESOLVE = 'return-resolve';
+const RETURN_REJECT = 'return-reject';
+const YIELD_RESOLVE = 'yield-resolve';
+const YIELD_REJECT = 'yield-reject';
+const messages = {
+	[RETURN_RESOLVE]: 'Prefer `return value` over `return Promise.resolve(value)`.',
+	[RETURN_REJECT]: 'Prefer `throw error` over `return Promise.reject(error)`.',
+	[YIELD_RESOLVE]: 'Prefer `yield value` over `yield Promise.resolve(value)`.',
+	[YIELD_REJECT]: 'Prefer `throw value` over `yield Promise.reject(value)`.',
+};
+
+const promiseResolveOrRejectSelector = methodCallSelector({
+	object: 'Promise',
+	methods: ['resolve', 'reject'],
+});
+const asyncArrowFunctionReturnSelector = `ArrowFunctionExpression[async=true] > ${promiseResolveOrRejectSelector}.body`;
+const returnStatementSelector = `ReturnStatement > ${promiseResolveOrRejectSelector}.argument`;
+const yieldExpressionSelector = `YieldExpression[delegate=false] > ${promiseResolveOrRejectSelector}.argument`;
+
+const functionTypes = new Set([
+	'ArrowFunctionExpression',
+	'FunctionDeclaration',
+	'FunctionExpression',
+]);
+const getParentFunction = node => {
+	let {parent} = node;
+	while (parent && !functionTypes.has(parent.type)) {
+		parent = parent.parent;
+	}
+
+	return parent;
+};
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => {
+	const sourceCode = context.getSourceCode();
+
+	/** @param {(isResolve: boolean, valueString: string, value) => import('eslint').Rule.ReportFixer} createFix */
+	const createProblem = (node, createFix, resolveMessage = RETURN_RESOLVE, rejectMessage = RETURN_REJECT) => {
+		const isResolve = node.callee.property.name === 'resolve';
+		const value = node.arguments.length === 1 ? node.arguments[0] : undefined;
+
+		let isInTryStatement = false;
+		let {parent} = node;
+		while (parent && !functionTypes.has(parent.type)) {
+			if (parent.type === 'TryStatement') {
+				isInTryStatement = true;
+				break;
+			}
+
+			parent = parent.parent;
+		}
+
+		return {
+			node: node.callee,
+			messageId: isResolve ? resolveMessage : rejectMessage,
+			fix: (!isResolve && isInTryStatement) || node.arguments.length > 1
```

Comment:
We need also check the only argument's type, it can't be `Promise.resolve(...args)`